### ISSUE TITLE
Add tensorboard support for segmentation metrics.

### ISSUE THREAD

**Issue by Abdul-Mukit** [roboflow/rf-detr#411]
### Search before asking

- [x] I have searched the RF-DETR [issues](https://github.com/roboflow/rf-detr/issues) and found no similar feature requests.

### Description

While training RFDETRSegPreview, the plots in tensorboard still only contain metrics related to BBox. Plots of segmentation metrics are missing.
In order to support this, the class MetricsTensorBoardSink needs to be enhanced. 

Expected behavior when training segmentation models:  

<img width="1678" height="543" alt="Image" src="https://github.com/user-attachments/assets/2de308e8-75c7-4de8-b173-a99f523752d0" />  

<img width="1790" height="1219" alt="Image" src="https://github.com/user-attachments/assets/4eaffcd9-190b-4a12-afe3-cb4152f9c8ef" />  

<img width="1865" height="1202" alt="Image" src="https://github.com/user-attachments/assets/9e919297-ebc6-4b41-9ffb-798045a10e4d" />

### Use case

Tensorboard logging of the mask metrics is needed to track and monitor experiments.

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by Borda** [comment]
@Abdul-Mukit Thanks for the request and the PR effort!

In v1.6.0, we've completely rebuilt the metrics pipeline using PyTorch Lightning + torchmetrics. Segmentation metrics (mask AP/AR) are now logged to TensorBoard alongside bbox metrics automatically when training segmentation models.

The RC is already available for testing. Closing this since the feature is now built-in. If you notice any seg metrics missing from TensorBoard on 1.6.0, please reopen or file a new issue!

**Comment by Abdul-Mukit** [comment]
Got it. Thank you.
